core/logger.go

```diff
@@ -25,29 +25,41 @@ type Logger interface {
 	Debug(...interface{})
 	// Debugln logs a message at level Debug.
 	Debugln(...interface{})
+	// Debugf formatted logs a message at level Debug.
+	Debugf(string, ...interface{})
 
 	// Info logs a message at level Info.
 	Info(...interface{})
 	// Infoln logs a message at level Info.
 	Infoln(...interface{})
```

Comment:
Do we really need *ln() versions of logging?
May be we should remove them and make other versions to print new line implicitly and obligatory?

Reply:
Debug() and Debugln() has the same behavior, but interface has both methods for compatibility

Reply:
If behaviors are the same, it's a little bit confusing to have two functions